### PR TITLE
Upgrade to Acquia Cloud API 2

### PR DESCRIPTION
**Pull request by weitzman**
- Depends on https://github.com/massgov/DS-Infrastructure/pull/189. Lets merge that first.

**Jira:**
https://jira.mass.gov/browse/DP-16283

**To Test:**
- [x] Get new drush aliases file from DS-Infrastructure. See https://github.com/massgov/DS-Infrastructure/pull/189
- [x] Populate 2 new .env variables. See env.example change
- [x] Push this branch to feature5 using ma:deploy. Assure it succeeds.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: composer.json
Comment (rbayliss) [inline composer.json]: Can we constrain this to stable releases, or is there a specific reason we need to be on dev? If there isn't a particular reason, how about if we switch to `~1.1`?

**Review by rbayliss — CHANGES_REQUESTED** [review]
This looks awesome!  My only request is to use a stable version of acquia-php-sdk-v2.

**Comment by rbayliss** [comment]
I did check that the aliases were right, that this deploys cleanly, and that `ma:latest-backup-url` still works.  Everything checks out, but I still think we might want to wait until after the holidays to merge so we don't put ourselves in a position of having nobody around to support a problematic release.  But I'm good to go with these changes as soon as folks are back.

File: composer.json
Comment (weitzman) [reply composer.json]: There is no release yet with https://github.com/typhonius/acquia-php-sdk-v2/pull/36 in it.

Comment (typhonius) [reply composer.json]: I'm planning on releasing a new tag for both 1.1 and 2.0 branches over the holiday period if that would help!

Comment (typhonius) [reply composer.json]: 1.1.1 and 2.0.1 have been released if you'd like to pin to those. Both have notifications built in so should be of use in this PR.

I'll mostly be working in the 2.x branch now as it's been rewritten to fulfil the growing needs of the library. That being said, because it has been rewritten, there were major API changes so would not apply to this PR without alterations.

Comment (rbayliss) [reply composer.json]: @typhonius, thanks for the update!  Very helpful.

Comment (yriahi) [reply composer.json]: I have now required `~1.1`. Thank you!

**Comment by rbayliss** [comment]
Ah - actually, I just realized that we need to update `docker-compose.yml` and `.circleci/mysql-rebuild/docker-compose.yml` to use the new environment variables. Let's keep both the `AC_API_*` and `AC_API2_*` variables in there and clean up the `AC_API_*` ones in a followup.

**Comment by yriahi** [comment]
@rbayliss I made the change to preserve both environment variables - Acquia Cloud v1 and v2.

**Pull request merged**
